src/main/java/org/victorrobotics/dtlib/subsystem/swerve/SwerveDrive.java

```diff
@@ -114,29 +112,17 @@ public final void setStates(SwerveModuleState... states) {
           + modules.length + " modules");
     }
 
-    double minCosine = 1;
     for (int i = 0; i < modules.length; i++) {
-      states[i] = SwerveModuleState.optimize(states[i], modules[i].getSteerAngle());
-      double cosine = modules[i].getSteerAngle()
-                                .minus(states[i].angle)
-                                .getCos();
-      if (cosine < minCosine) {
-        minCosine = cosine;
-      }
-    }
-    minCosine *= minCosine * minCosine; // minCosine to the 3rd power
-    for (int i = 0; i < modules.length; i++) {
-      states[i].speedMetersPerSecond *= minCosine;
```

Comment:
IIRC we originally added a correction like this to limit the speed of wheels that are facing the wrong way, until the motors align them a bit more with the direction of travel (though I don't recall the cubing). We'll need to do some testing as soon as we have a swerve chassis and shake all the issues out of this implementation before the build season.

Reply:
That's my plan for the new swerve chassis we just got